### PR TITLE
Support rollback nodes in TransactionIndexing

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/test/suite/scala/platform/store/dao/events/TransactionIndexingSpec.scala [hunk 1]
```diff
@@ -40,5 +45,125 @@ final class TransactionIndexingSpec extends AnyWordSpec with Matchers {
           .ContractWitnessesInfo(Set.empty, aDivulgence)
       )
     }
+
+    "filter out rollback nodes" in {
+      val partyStr = "Alice"
+      val party = Party.assertFromString(partyStr)
+      val bobStr = "Bob"
+      val bob = Party.assertFromString(bobStr)
+//    val offset = LedgerOffset.Absolute(LedgerString.assertFromString("offset"))
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/store/dao/events/TransactionIndexingSpec.scala:54]: Can you remove this commented line?

**Comment by tudor-da** [comment]
This implementation targets the legacy index schema, while we are already working on the new schema (in package `appendonlydao`. I am not sure if it should be you to implement it in the upcoming schema. But if we decide to port it later, we need to keep track of it. @mziolekda @nmarton-da what do you think?

File: ledger/participant-integration-api/src/test/suite/scala/platform/store/dao/events/TransactionIndexingSpec.scala (hunk 1 above)
Comment (cocreature) [reply ledger/participant-integration-api/src/test/suite/scala/platform/store/dao/events/TransactionIndexingSpec.scala]: 6dff0ee6d9b8156e816de569e1fd4e10ab5d1a7b

**Comment by cocreature** [comment]
> This implementation targets the legacy index schema, while we are already working on the new schema (in package `appendonlydao`. I am not sure if it should be you to implement it in the upcoming schema. But if we decide to port it later, we need to keep track of it. @mziolekda @nmarton-da what do you think?

Where is the new schema code for this? Contrary to some other files, this only seems to exist once. We need exception support everywhere so this absolutely needs to be ported to the new schema. 

I care less about who has to do the work and more that we get it working so happy to take a shot or help someone if you can point me to the right place.

I think it probably makes sense to keep that for a separate PR though.

**Comment by tudor-da** [comment]
> > This implementation targets the legacy index schema, while we are already working on the new schema (in package `appendonlydao`. I am not sure if it should be you to implement it in the upcoming schema. But if we decide to port it later, we need to keep track of it. @mziolekda @nmarton-da what do you think?
> 
> Where is the new schema code for this? Contrary to some other files, this only seems to exist once. We need exception support everywhere so this absolutely needs to be ported to the new schema.
> 
> I care less about who has to do the work and more that we get it working so happy to take a shot or help someone if you can point me to the right place.
> 
> I think it probably makes sense to keep that for a separate PR though.

Ok, now it is more clear. It seems the [new code](https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/UpdateToDBDTOV1.scala#L140-L263) would be already ignoring Rollback nodes. So nothing is needed in that area, I believe. Thanks, @moritzkiefer-da

**Review by tudor-da — APPROVED** [review]

**Comment by cocreature** [comment]
> Ok, now it is more clear. It seems the [new code](https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/UpdateToDBDTOV1.scala#L140-L263) would be already ignoring Rollback nodes. So nothing is needed in that area, I believe. Thanks, @moritzkiefer-da

Looking at this, I’m not entirely sure that’s right.I think you filter out rollback nodes but you will still get a create under a rollback node. I’ll make sure to look into this and either fix it myself or reach out to you if I need help.

**Pull request merged**